frontend/src/features/documents/documentUpload/DocumentUploadView.tsx

```diff
@@ -147,8 +146,9 @@ const DocumentUploadView: React.FunctionComponent<IDocumentUploadViewProps> = pr
                       required={value.required === true}
                       onChange={event => {
                         if (value.metadata_type?.id) {
-                          props.documentMetadata[value.metadata_type?.id.toString()] =
-                            event.target.value;
+                          formikProps.values.mayanDocumentMetadata[
```

Comment:
still not sure this is necessary. Formik uses the field parameter to set the value when it detects a change so this would not be necessary unless there is something different about the input field. This could be related to the field parameter on the previous line. That would be more obvious if the initial formik data would be populated. @devinleighsmith you have more experience with formik, what do you think?

Reply:
Correct, this isn't a valid pattern. if an input is a list element it should be named appropriately with the id. for example, the first input would be mayanDocumentMetadata.`${value.metadata_type?.id.toString()}` or similar. Have a look at https://formik.org/docs/api/fieldarray for a related example

Reply:
Thanks Devin , formik is already adding all fields directly in the values. Changed the code to use those directly.